### PR TITLE
Feature mentor apply link

### PR DESCRIPTION
**Pull request by tyleryy**
### Overview

Added link to mentor application google form.

---

Reference to form [here](https://forms.gle/fwWJtsQ3GXQFGjS47)

### CODE REVIEW

**Comment by tyleryy** [comment]
I also added a `/mentor` redirect route to the form. Let me know if this is desired behavior.

**Comment by tyleryy** [comment]
Actually, on the redirect, I noticed the web page renders empty first then redirects to the google form.

**Comment by tyleryy** [comment]
Blocked by apply link PR, will pull to get redirect urls once merged

**Review by samderanova — CHANGES_REQUESTED** [review]
One small issue and we're good to go! Thanks for working on this as well.

File: apps/site/src/app/mentor/page.tsx
Comment (tyleryy) [reply apps/site/src/app/mentor/page.tsx:1]: yea it was blocked by the other PR. Now that it is merged Ill make the change now! :)

Comment (tyleryy) [reply apps/site/src/app/mentor/page.tsx:1]: kk changes made

**Comment by tyleryy** [comment]
Ready for Review!

**Review by alexanderl19 — APPROVED** [review]

**Comment by taesungh** [comment]
Force unmerged given messiness of tree, reopening as #43.

**Pull request merged**
